cluster/gce/win1803/k8s-node-setup.psm1

```diff
@@ -995,6 +1006,23 @@ function Start-WorkerServices {
   # Figure out how to change the directory that the kubelet monitors for new
   # pod manifests.
 
+
+
+  # We configure the service to restart on failure, after 10s wait. We reset the restart
+  # count to 0 each time, so we re-use our restart/10000 action on each failure.
+  # Note it currently restarts even when explicitly stopped, you have to delete the service
+  # entry to *really* kill it (e.g. `sc.exe delete kubelet`). See issue #72900.
+  if (Get-Process | Where-Object Name -eq "kubelet") {
+    Log-Output `
+        "A kubelet process is already running, don't know what to do" `
+        -Fatal
+  }
+  Log-Output "Creating kubelet service"
+  sc.exe create kubelet binPath= "${env:NODE_DIR}\kubelet.exe ${kubelet_args}"
+  sc.exe failure kubelet reset= 0 actions= restart/10000
+  Log-Output "Starting kubelet service"
+  sc.exe start kubelet
```

Comment:
What will happen here if we reboot the Windows node after setting it up the first time? This looks probably fine but can you confirm that:

1. The kubelet and kube-proxy services won't auto-restart on boot (I expect we want to just restart them manually here);
1. These sc.exe commands won't cause the script to exit if the service has already been registered in the past.

Reply:
Let me try rebooting a node and see what happens :). I think it will be fine, though, as I got the commands wrong several times when figuring this out, and the scripts just kept running.

The default for sc.exe is to configure services with `start= demand`, which means it must be started manually.

Reply:
Reboot worked fine. We got the expected failure message:
```
[SC] CreateService FAILED 1073:

The specified service already exists.
```
for both kubelet and kube-proxy, and the startup script kept running as expected. Smoke test still passes after reboot.

I also updated the script to explicitly use `start= demand` to make it extra-clear.